src/Microsoft.Diagnostics.DebugServices.Implementation/CommandService.cs

```diff
@@ -233,10 +230,7 @@ private void CreateCommand(Type type, CommandAttribute commandAttribute, Func<IS
             _rootBuilder.AddCommand(command);
         }
 
-        private void BuildParser()
-        {
-            _parser = _rootBuilder.Build();
-        }
+        private Parser Parser => _parser ??= _rootBuilder.Build();
```

Comment:
```suggestion
        private Parser Parser
        {
            get { _parser ??= _rootBuilder.Build(); }
            set { value is null ? _parser = null : throw new InvalidOperationException($"Explicit setting should never be done. Use helper methods in {nameof(CommandService)} to add them appropriately."); }
```

We could make the exception debug only if we fear it might slip into release builds.

Reply:
Doesn't this seem a little overkill for a private property/field?  What about a private function called "FlushParser() => _parser = null;" right next to the property?

Reply:
The problem is more you're still not validating that no one accesses anything. If we don't want exceptions - that makes sense. But having a property that's private the class with a backing field is not doing much. I guess if we can't really restrict access from the class, we can leave as is.